### ISSUE TITLE
Problems with --with-*install_dir

### ISSUE THREAD

**Issue by mythi** [intel/QAT_Engine#82]
1. The readme talks about ```--with-qat_build_dir``` but it should be ```--with-qat_install_dir```

2. ```--with-openssl_install_dir``` is buggy. It cannot be used on Debian based distros where openssl engines dir is ```/usr/lib/<build-arch>/engines-1.1```.

**Comment by stevelinsell** [comment]
Hi @mythi,

Thankyou so much for the valuable feedback.

Good spot for the README.md, this is definitely wrong and I shall correct it in a future release.

With regard to the second point, I believe that the `/usr/lib/<build-arch>/engines-1.1` directory format is only used in Debian repackaged OpenSSL distributions, i.e if you build OpenSSL from the github repo source code under a Debian based distribution you won't get this directory structure by default. The assumption in the past has always been that people would be building directly from the github source code, but perhaps that is an assumption we should not be making. I'll have a think about how we can deal with this, I suspect it will require an additional configuration option for the Debian distribution case.

Kind Regards,

Steve.

**Comment by stevelinsell** [comment]
@mythi , I agree, the container route is an option more and more people are looking at taking, and being able to just take a base image is easiest.
Which reminds me, I must add a link in the README.md to the 01.org documentation on running QAT in a Linux Container:
[Using Intel® QuickAssist Technology in Linux Container and Docker](https://01.org/sites/default/files/downloads/intelr-quickassist-technology/337020-001qatwcontaineranddocker.pdf)

`--with-openssl_engine_install_dir` is definitely an option, I'll need to look whether that will work in all cases. If we can do it without increasing the amount of configuration options but just by replacing an existing one then that would definitely be preferable. 

Kind Regards,

Steve.

**Comment by mythi** [comment]
@stevelinsell the doc provides good input but could use an update. The pre-assumption for Docker containers: _Make sure Nginx and OpenSSL have been configured properly in the host._ is not justified in container world. Instead, These should be installable through a ```Dockerfile```.  I have that WIP and can share it later.

**Comment by mythi** [comment]
@stevelinsell I can see this these are fixed in .39. Thanks!

One minor comment/feedback about the "Debian install": with ```make install``` the intermediate ```libqat.so``` is now installed in ```/usr/lib/x86_64-linux-gnu/lib``` and the final ```qat.so``` gets copied to the right place. It'd be nice to avoid that intermediate install.
